perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_member_attribute_def_def_o365EmailAddresses_mu.java

```diff
@@ -53,31 +58,39 @@
 	private static final String NAMESPACE = AttributesManager.NS_MEMBER_ATTR_DEF;
 	static final String UCO_ATTRIBUTE = AttributesManager.NS_USER_ATTR_DEF + ":login-namespace:mu";
 
-	@Override
-	public void checkAttributeSemantics(PerunSessionImpl sess, Member member, Attribute attribute) throws InternalErrorException, WrongAttributeValueException, WrongReferenceAttributeValueException, WrongAttributeAssignmentException {
-		log.trace("checkAttributeSemantics(member={},attribute={})", member, attribute);
-		ArrayList<String> emails;
-
-		//get values
+	public void checkAttributeSyntax(PerunSessionImpl perunSession, Member member, Attribute attribute) throws WrongAttributeValueException {
 		Object value = attribute.getValue();
-		if (value == null) {
-			throw new WrongAttributeValueException(attribute, member, "can't be null.");
-		} else if (!(value instanceof ArrayList)) {
+		List<String> emails;
+
+		if (value == null) return;
+		else if (!(value instanceof ArrayList)) {
 			throw new WrongAttributeValueException(attribute, member, "is of type " + value.getClass() + ", but should be ArrayList");
 		} else {
-			//noinspection unchecked
-			emails = (ArrayList<String>) value;
+			emails = attribute.valueAsList();
 		}
-		//check syntax of all values
+
 		for (String email : emails) {
 			Matcher emailMatcher = emailPattern.matcher(email);
 			if (!emailMatcher.matches())
 				throw new WrongAttributeValueException(attribute, member, "Email " + email + " is not in correct form.");
 		}
+	}
+
+	@Override
+	public void checkAttributeSemantics(PerunSessionImpl sess, Member member, Attribute attribute) throws InternalErrorException, WrongReferenceAttributeValueException, WrongAttributeAssignmentException {
+		log.trace("checkAttributeSemantics(member={},attribute={})", member, attribute);
+		List<String> emails;
+
+		//get values
+		if (attribute.getValue() == null) {
+			throw new WrongReferenceAttributeValueException(attribute, "can't be null.");
+		} else {
+			emails = attribute.valueAsList();
+		}
 
 		//check for duplicities
 		if (hasDuplicate(emails)) {
-			throw new WrongAttributeValueException(attribute, member, "duplicate values");
+			throw new WrongReferenceAttributeValueException(attribute, "has duplicate values");
```

Comment:
This looks like syntax check. Syntax says that duplicity in values is not allowed.

Reply:
Yes, it should be syntax check now.